File: Lista-15/python/12.py
class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outro):
        return self.preco + outro.preco

# Exemplo de uso
produto1 = Produto("Produto 1", 100)
produto2 = Produto("Produto 2", 150)
print(produto1 + produto2)  # Saída: 250

File: Lista-15/java/12.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public double somar(Produto outro) {
        return this.preco + outro.preco;
    }

    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto 1", 100);
        Produto produto2 = new Produto("Produto 2", 150);
        System.out.println(produto1.somar(produto2));  // Saída: 250
    }
}
